exercises/practice/split-second-stopwatch/.meta/src/SplitSecondStopwatch.example.elm

```diff
@@ -0,0 +1,141 @@
+module SplitSecondStopwatch exposing
+    ( State(..)
+    , Stopwatch
+    , advanceTime
+    , currentLap
+    , lap
+    , new
+    , previousLaps
+    , reset
+    , start
+    , state
+    , stop
+    , total
+    )
+
+
+type State
+    = Ready
+    | Running
+    | Stopped
+
+
+type Stopwatch
+    = Stopwatch
+        { state_ : State
+        , currentLap_ : Int
+        , previousLaps_ : List Int
+        }
+
+
+new : Stopwatch
+new =
+    Stopwatch { state_ = Ready, currentLap_ = 0, previousLaps_ = [] }
+
+
+state : Stopwatch -> State
+state (Stopwatch { state_ }) =
+    state_
+
+
+currentLap : Stopwatch -> String
+currentLap (Stopwatch { currentLap_ }) =
+    formatTime currentLap_
+
+
+previousLaps : Stopwatch -> List String
+previousLaps (Stopwatch { previousLaps_ }) =
+    previousLaps_
+        |> List.reverse
+        |> List.map formatTime
+
+
+advanceTime : String -> Stopwatch -> Stopwatch
+advanceTime time (Stopwatch ({ state_, currentLap_ } as stopwatch)) =
+    case state_ of
+        Running ->
+            Stopwatch { stopwatch | currentLap_ = currentLap_ + parseTime time }
+
+        _ ->
+            Stopwatch stopwatch
+
+
+total : Stopwatch -> String
+total (Stopwatch { currentLap_, previousLaps_ }) =
+    (currentLap_ :: previousLaps_)
+        |> List.sum
+        |> formatTime
+
+
+start : Stopwatch -> Result String Stopwatch
+start (Stopwatch ({ state_ } as stopwatch)) =
```

Comment:
Maybe we could use a finite state machine here, instead of using Result (and in the other places)?

Reply:
I was trying to stick to the specs.
How would you write a finite state machine though? With phantom types? Or just by ignoring error instructions?

Reply:
I've seen `aState` used before, which fits with it being an instance of the state type

Reply:
I don't feel that strongly about it, but you could check what other people think. I suspect it won't be that easy to change it, and they might suggest to create a new exercise based on finite state machines (which would be cool).
In the mean time, I'll keep it this way.